### PR TITLE
Fix Typos

### PR DESCRIPTION
**Pull request by detinsley1s**
I fixed numerous typos I found within the document, such as spelling mistakes and punctuation mistakes.

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as `[TEXT](URL)`
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md [hunk 1]
```diff
@@ -644,10 +644,10 @@ There is one caveat with using some of the guards: Apple only provides them for
...
+Visual Studio offers a number of debugging aids for use during development. The aids are called [Managed Debugging Assistants (MDAs)](https://docs.microsoft.com/en-us/dotnet/framework/debug-trace-profile/diagnosing-errors-with-managed-debugging-assistants). You can find the MDAs on the _Debug_ menu, then _Exceptions_ submenu. MDAs allow you to tune your debugging experience by, for example, filtering exceptions for which the debugger should snap. For more details, see Stephen Toub's _[Let The CLR Find Bugs For You With Managed Debugging Assistants](https://docs.microsoft.com/en-us/archive/msdn-magazine/2006/may/let-the-clr-find-bugs-for-you-with-managed-debugging-assistants)_.
```
Comment (kwwall) [inline cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md:647]: I think 'aide' is the correct work here. From the American Heritage Dictionary, used as a noun:

- **aide**: An assistant; a helper. Example: _"A nurse's aide"._
- **aid**: The act or result of helping; assistance. Example: _"gave aid to the enemy"._

I think aide is a better fit here.

**Review by kwwall — DISMISSED** [review]
A few comments. Ideally, I'd like them addressed, but I'm not one to let the perfect be the enemy of the good. This is a great job overall and probably most of my comments are minor nitpicking.

File: cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md (hunk 1 above)
Comment (detinsley1s) [reply cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md:647]: The reason I changed it was an aide is a person who helps someone. The helpers in this context aren't human beings, which is why "aid" is the correct word here. [https://www.dictionary.com/e/aid-vs-aide/](url)

> Additionally, the word aide only refers to people, while aid can refer to people or objects.

**Comment by kwwall** [comment]
I wouldn't said 'aide' _only_ can refer to people, but it _typically_ does. And in the not-too-distant future, when IDEs start picking up AI assistance or we all start using smart, personalized AI agents, it wouldn't surprise me if we start referring to that personalized assistance as a 'debugging aide', just like we would a personal human mentor who might help us with debugging. But I digress. As I said, I'm okay with the way things are. Probably the whole `s/makefile/Makefile/` changes are the ones that are most upsetting to my nitpicking, because those go against common use of the 'make' man page. But in the end, it's not worth arguing over.

**Comment by mackowski** [comment]
@detinsley1s can you please change or argument your wording in lines where @kwwall left comments?

**Comment by kwwall** [comment]
Just to be clear, I only care about changing 'Makefile' to 'makefile'. The nomenclature ought to be in accordance with the **make(1)** manual page.

**Comment by detinsley1s** [comment]
Hi, I changed every instance of "Makefile" to "makefile," except in one instance, since it was the title of a section, and titles are capitalized. I hope that is acceptable. Thank you.

**Review by kwwall — APPROVED** [review]
Looks good to me. Thanks for making these changes @detinsley1s.

**Review by jmanico — APPROVED** [review]

**Pull request merged**